CustomScript/customscript.py

```diff
@@ -405,6 +417,8 @@ def to_process(file_path, extensions=['.sh', ".py"]):
     for extension in extensions:
         if file_path.endswith(extension):
             return True
+        else:
+            return False
     with open(file_path, 'rb') as f:
         contents = f.read(64)
```

Comment:
Why are you searching the first 64 characters? Near as I can tell, there are only two cases:
- File begins with a BOM. Strip the BOM, convert from UTF-16 (in the byte gender indicated by the BOM) to UTF-8, and convert /r/n to /n
- File begins with #!. Just convert \r\n to \n.

When I say "begins with", I really mean that - the absolute first characters of the file.

I'm still not a fan of this autoconversion; it will bite you eventually. But tightly restricting the scope in which you apply the conversion is your best defense.